### PR TITLE
WC migration to WalletKit

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes APP-1716
## What changed (plus any additional context for devs)

- Bumped all wc packages to latest version
- Migrated Web3Wallet to WalletKit package

The acceptable conditions are:
are:
- Latency for the connection request to show up in poor network (3G) conditions should be under 15 seconds. 
- Latency for the connection request to show up in poor network (3G) conditions should be under 10 seconds.

## Screen recordings / screenshots

https://github.com/user-attachments/assets/08091f81-3f0e-42fc-8329-ee860d1f74ec

## What to test
WC v2

### CODE REVIEW

File: src/walletConnect/index.tsx [hunk 1]
```diff
@@ -323,11 +323,10 @@ async function rejectProposal({
 }
 
 // listen for THIS topic pairing, and clear timeout if received
-function trackTopicHandler(proposal: Web3WalletTypes.SessionProposal | Web3WalletTypes.AuthRequest) {
+function trackTopicHandler(proposal: WalletKitTypes.SessionProposal) {
   logger.debug(`[walletConnect]: pair: handler`, { proposal });
 
-  const { metadata } =
-    (proposal as Web3WalletTypes.SessionProposal).params.proposer || (proposal as Web3WalletTypes.AuthRequest).params.requester;
+  const { metadata } = (proposal as WalletKitTypes.SessionProposal).params.proposer;
```
Comment (walmat) [inline src/walletConnect/index.tsx:329]:
```suggestion
  const { metadata } = proposal.params.proposer;
```

File: src/walletConnect/index.tsx [hunk 2]
```diff
@@ -533,6 +543,8 @@ export async function onSessionProposal(proposal: Web3WalletTypes.SessionProposa
                 });
               }
             } else {
+              console.log('[WC DEBUG] rejected proposal');
```
Comment (walmat) [inline src/walletConnect/index.tsx:546]: use logger or remove if not needed

File: src/walletConnect/index.tsx [hunk 3]
```diff
@@ -516,6 +524,8 @@ export async function onSessionProposal(proposal: Web3WalletTypes.SessionProposa
                 namespaces: namespaces.result,
               });
 
+              console.log('[WC DEBUG] session approved');
```
Comment (walmat) [inline src/walletConnect/index.tsx:527]: use logger or remove if not needed

File: src/walletConnect/index.tsx [hunk 4]
```diff
@@ -468,10 +467,15 @@ export async function onSessionProposal(proposal: Web3WalletTypes.SessionProposa
       verifiedData,
       timedOut: false,
       callback: async (approved, approvedChainId, accountAddress) => {
-        const client = await getWeb3WalletClient();
+        console.log('[WC DEBUG] executing callback');
```
Comment (walmat) [inline src/walletConnect/index.tsx:470]: use logger or remove if not needed

File: src/walletConnect/index.tsx [hunk 5]
```diff
@@ -468,10 +467,15 @@ export async function onSessionProposal(proposal: Web3WalletTypes.SessionProposa
...
+        const client = await getWalletKitClient();
+        console.log('[WC DEBUG] got client');
```
Comment (walmat) [inline src/walletConnect/index.tsx:472]: use logger or remove if not needed

File: src/walletConnect/index.tsx [hunk 6]
```diff
@@ -468,10 +467,15 @@ export async function onSessionProposal(proposal: Web3WalletTypes.SessionProposa
...
+
         const { id, proposer, requiredNamespaces } = proposal.params;
 
         if (approved) {
+          console.log('[WC DEBUG] approved true');
```
Comment (walmat) [inline src/walletConnect/index.tsx:477]: use logger or remove if not needed

File: src/walletConnect/index.tsx [hunk 7]
```diff
@@ -502,7 +506,11 @@ export async function onSessionProposal(proposal: Web3WalletTypes.SessionProposa
           logger.debug(`[walletConnect]: session approved namespaces`, { namespaces }, logger.DebugContext.walletconnect);
 
           try {
+            console.log('[WC DEBUG] try');
```
Comment (walmat) [inline src/walletConnect/index.tsx:509]: use logger or remove if not needed

File: src/walletConnect/index.tsx [hunk 8]
```diff
@@ -502,7 +506,11 @@ export async function onSessionProposal(proposal: Web3WalletTypes.SessionProposa
...
+
             if (namespaces.success) {
+              console.log('[WC DEBUG] namespaces success');
```
Comment (walmat) [inline src/walletConnect/index.tsx:512]: use logger or remove if not needed

File: src/walletConnect/index.tsx [hunk 9]
```diff
@@ -546,6 +558,8 @@ export async function onSessionProposal(proposal: Web3WalletTypes.SessionProposa
               });
             }
           } catch (e) {
+            console.log('[WC DEBUG] catch', e);
```
Comment (walmat) [inline src/walletConnect/index.tsx:561]: use logger or remove if not needed

File: src/walletConnect/index.tsx [hunk 10]
```diff
@@ -566,6 +580,8 @@ export async function onSessionProposal(proposal: Web3WalletTypes.SessionProposa
             });
           }
         } else if (!approved) {
+          console.log('[WC DEBUG] not approved');
```
Comment (walmat) [inline src/walletConnect/index.tsx:583]: use logger or remove if not needed

File: src/walletConnect/index.tsx [hunk 11]
```diff
@@ -829,10 +845,12 @@ export async function handleSessionRequestResponse(
   store.dispatch(removeRequest(sessionRequestEvent.id));
 }
 
-export async function onAuthRequest(event: Web3WalletTypes.AuthRequest) {
+export async function onSessionAuthenticate(event: WalletKitTypes.SessionAuthenticate) {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore Can be fixed once we wipe wc v1 from the codebase
```
Comment (walmat) [inline src/walletConnect/index.tsx:850]: is this legacy now?

File: src/walletConnect/index.tsx [hunk 12]
```diff
@@ -845,6 +863,8 @@ export async function onAuthRequest(event: Web3WalletTypes.AuthRequest) {
 
       // exit early if possible
       if (selectedWallet?.type === WalletTypes.readOnly) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore Can be fixed once we wipe wc v1 from the codebase
```
Comment (walmat) [inline src/walletConnect/index.tsx:867]: is this legacy now?

File: src/walletConnect/index.tsx [hunk 13]
```diff
@@ -875,7 +895,8 @@ export async function onAuthRequest(event: Web3WalletTypes.AuthRequest) {
 
           return undefined;
         }
-
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore Can be fixed once we wipe wc v1 from the codebase
```
Comment (walmat) [inline src/walletConnect/index.tsx:899]: is this legacy now?

File: src/walletConnect/index.tsx [hunk 14]
```diff
@@ -929,7 +952,7 @@ export async function onAuthRequest(event: Web3WalletTypes.AuthRequest) {
 
   // need to prefetch dapp metadata since portal is static
   const url =
-    // @ts-ignore Web3WalletTypes.AuthRequest type is missing VerifyContext
+    // @ts-ignore WalletKitTypes.AuthRequest type is missing VerifyContext
     event?.verifyContext?.origin || event.params.requester.metadata.url;
```
Comment (walmat) [inline src/walletConnect/index.tsx:956]: this is `event?.verifyContext?.verified?.origin`

File: src/walletConnect/index.tsx [hunk 15]
```diff
@@ -939,7 +962,7 @@ export async function onAuthRequest(event: Web3WalletTypes.AuthRequest) {
       AuthRequest({
         authenticate,
         requesterMeta: event.params.requester.metadata,
-        // @ts-ignore Web3WalletTypes.AuthRequest type is missing VerifyContext
+        // @ts-ignore WalletKitTypes.AuthRequest type is missing VerifyContext
         verifiedData: event?.verifyContext,
```
Comment (walmat) [inline src/walletConnect/index.tsx:966]: same here. use `event?.verifyContext?.verified`

File: src/walletConnect/index.tsx [hunk 16]
```diff
@@ -904,6 +925,8 @@ export async function onAuthRequest(event: Web3WalletTypes.AuthRequest) {
       }
 
       // respond to WC
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore Can be fixed once we wipe wc v1 from the codebase
```
Comment (walmat) [inline src/walletConnect/index.tsx:929]: is this legacy now?

**Review by walmat — CHANGES_REQUESTED** [review]
few comments. I think we need to explore what to do with all of the `@ts-ignore` in this file

File: src/walletConnect/index.tsx (hunk 14 above)
Comment (walmat) [reply src/walletConnect/index.tsx:956]: can also remove the `@ts-ignore` after you make that change

File: src/walletConnect/index.tsx (hunk 15 above)
Comment (walmat) [reply src/walletConnect/index.tsx:966]: and remove `@ts-ignore`

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@bruno/wc-walletkit/57150b05e769c078754af0aa6d137f598f5589b8.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@bruno/wc-walletkit/57150b05e769c078754af0aa6d137f598f5589b8.ipa) for 57150b05e769c078754af0aa6d137f598f5589b8

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@bruno/wc-walletkit/afba60528a03aa2fd9ebcc7d5317deab3356782b.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@bruno/wc-walletkit/afba60528a03aa2fd9ebcc7d5317deab3356782b.ipa) for afba60528a03aa2fd9ebcc7d5317deab3356782b

**Review by walmat — APPROVED** [review]

**Review by derHowie — APPROVED** [review]

**Pull request merged**